pkg/kubelet/hyper/hyper.go

```diff
@@ -812,18 +885,12 @@ func (r *runtime) RunPod(pod *api.Pod, restartCount int, pullSecrets []api.Secre
 	}
 
 	// Create and start hyper pod
-	podSpec, err := r.getPodSpec(podFullName)
```

Comment:
So the workflow is: `buildHyperPod -> podSpec -> json file -> kubestack change the file -> podSpec -> createPod`

Be careful when doing marshal and unmarshal .